### PR TITLE
Landing page content for authentified users

### PR DESCRIPTION
**Pull request by nworr**
the administrator can now define content at the top and bottom of the landing page that will only be displayed to authenticated users, with an option to display unauthenticated content or not 

 
image

Ticket : # 

Funded by JPEE

### CODE REVIEW

**Comment by gioman** [comment]
Nice, could be possible in the WYSIWYG editor enable the button to allow see/edit the html code?

**Review by laurentj — APPROVED** [review]
LGTM

**Comment by laurentj** [comment]
> Nice, could be possible in the WYSIWYG editor enable the button to allow see/edit the html code?

I'm not sure we would want this feature. Most of users don't know HTML, and could write bad html which could break the web page. As we manage hundred Lizmap instances, we don't want to manage such problems on our customer support service ;-). Or this feature must be activated with a configuration parameter.

**Comment by gioman** [comment]
> I'm not sure we would want this feature. Most of users don't know HTML, and could write bad html which could break the web page. As we manage hundred Lizmap instances, we don't want to manage such problems on our customer support service ;-). Or this feature must be activated with a configuration parameter.

@laurentj fair enough. Could you point me where I can patch my local installations in order to enable the html code editor? i have looked into the LMWC code but have not figure where the important bits are.

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 1]
```diff
@@ -367,8 +367,15 @@ project.modal.button.close=Close
 project.list.no.hidden.column.content=No extra content for this project in the following hidden table columns
 
 
-landingPageContent.top=Content at the top of the landing page
-landingPageContent.bottom=Content at the bottom of the landing page
+landingPageContent.authed=Content for authentified users
+landingPageContent.unauthed=Content for unauthentified users
+landingPageContent.unauthed.position=Add content for unauthentified user to authentified users ?
```
Comment (Gustry) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:372]:
```suggestion
landingPageContent.unauthed.position=Add content for unauthentified user to authentified users
```

Usually HIG guidelines don't include question marks. I tried to look some rules, look in your OS settings or in QGIS
https://www.drupal.org/docs/develop/user-interface-standards/radio-buttons

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 2]
```diff
@@ -367,8 +367,15 @@ project.modal.button.close=Close
...
+landingPageContent.group.top=Top of the landing page
+landingPageContent.group.bottom=Bottom of the landing page
+landingPageContent.unauthed.position.before=before content for authentified
+landingPageContent.unauthed.position.after=after content for authentified
+landingPageContent.unauthed.position.no=no
```
Comment (Gustry) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:377]:
```suggestion
landingPageContent.unauthed.position.before=Before content for authentified
landingPageContent.unauthed.position.after=After content for authentified
landingPageContent.unauthed.position.no=No
```
Same, HIG guidelines, radio buttons always start with capital letters

**Review by Gustry — COMMENTED** [review]
Please switch `authentified` to `authenticated`

https://writingtips.org/authentication-or-authentification/

**Review by Gustry — APPROVED** [review]
For the UX : LGTM

**Review by rldhont — APPROVED** [review]

**Comment by nworr** [comment]
@gioman the editor "CKEditor" is provied by jelix, but it seems you can "live" edit its configuration in the file
lizmap\www\assets\jelix\js\jforms\htmleditors\ckeditor_ckdefault.js (it seem the "default" is used in most case) for example by adding `sourceEditing` feature https://ckeditor.com/docs/ckeditor5/latest/features/source-editing.html

Warning : this file will be overwritten to its original content on each update

**Pull request merged**